src/test/run-make/issue-11908/Makefile

```diff
@@ -13,11 +13,11 @@ all:
 	mv $(call DYLIB,foo) $(TMPDIR)/other
 	$(RUSTC) foo.rs --crate-type=dylib
 	$(RUSTC) bar.rs -L $(TMPDIR)/other 2>&1 | \
-		grep "multiple dylib candidates"
+		grep "assertion failed" && exit 1 || exit 0
```

Comment:
I think these tests should pass, right? With your changes the SVH is the same so they should get deduplicated so the compilation should succeed?

I'm a little wary to have a test that the compiler doesn't print "assertion failed" as that's a *very* brittle test to easily break without actually detecting a regression, but if these compiles are successful maybe that part can be tested?